From 19e5ddd1ee10e86fc428dce2bd2c5f813cd65a3c Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 17 Sep 2020 20:14:36 +0800
Subject: [PATCH] arm64: rockchip_defconfig: merge r/android-4.19 android base
 config

https://android.googlesource.com/kernel/configs
commit 5f72fe526dfe ("Merge "Require CONFIG_ANDROID_BINDERFS for R launching devices." into rvc-dev")

android-base.config and android-base-conditional.xml:

+CONFIG_IKHEADERS=m
-CONFIG_RT_GROUP_SCHED=y
-# CONFIG_UTS_NS is not set
+CONFIG_BPF_JIT_ALWAYS_ON=y
-CONFIG_MODULE_FORCE_UNLOAD=y
+CONFIG_NETFILTER_XT_MATCH_QUOTA2_LOG=y
-CONFIG_IP6_NF_NAT=y
-CONFIG_IP6_NF_TARGET_MASQUERADE=y
+CONFIG_BPF_JIT=y
+# CONFIG_FW_CACHE is not set
+CONFIG_DM_SNAPSHOT=y
+CONFIG_DUMMY=y
+CONFIG_VETH=y
+CONFIG_EXT4_FS_POSIX_ACL=y
-# CONFIG_F2FS_FS_POSIX_ACL is not set
+CONFIG_FS_VERITY=y
+CONFIG_FS_VERITY_BUILTIN_SIGNATURES=y
+CONFIG_STATIC_USERMODEHELPER=y
+CONFIG_STATIC_USERMODEHELPER_PATH=""

Except CONFIG_ANDROID_BINDERFS=y which do not supported by Android before 11.
CONFIG_ANDROID_BINDERFS is set by Android.

Signed-off-by: Tao Huang <huangtao@rock-chips.com>
Change-Id: Ic63cfa0d0294ddc7c623c44d5a183979eed74b9c
---
 arch/arm64/configs/rockchip_defconfig | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/configs/rockchip_defconfig b/arch/arm64/configs/rockchip_defconfig
index 2ee0b5e951ab..c7821343ea4d 100644
--- a/arch/arm64/configs/rockchip_defconfig
+++ b/arch/arm64/configs/rockchip_defconfig
@@ -12,19 +12,18 @@ CONFIG_TASK_IO_ACCOUNTING=y
 CONFIG_PSI=y
 CONFIG_IKCONFIG=y
 CONFIG_IKCONFIG_PROC=y
+CONFIG_IKHEADERS=m
 CONFIG_LOG_BUF_SHIFT=19
 CONFIG_CGROUPS=y
 CONFIG_MEMCG=y
 CONFIG_MEMCG_SWAP=y
 CONFIG_BLK_CGROUP=y
 CONFIG_CGROUP_SCHED=y
-CONFIG_RT_GROUP_SCHED=y
 CONFIG_CGROUP_FREEZER=y
 CONFIG_CPUSETS=y
 CONFIG_CGROUP_CPUACCT=y
 CONFIG_CGROUP_BPF=y
 CONFIG_NAMESPACES=y
-# CONFIG_UTS_NS is not set
 # CONFIG_PID_NS is not set
 CONFIG_SCHED_TUNE=y
 CONFIG_BLK_DEV_INITRD=y
@@ -38,6 +37,7 @@ CONFIG_INITRD_ASYNC=y
 # CONFIG_FHANDLE is not set
 CONFIG_KALLSYMS_ALL=y
 CONFIG_BPF_SYSCALL=y
+CONFIG_BPF_JIT_ALWAYS_ON=y
 CONFIG_EMBEDDED=y
 # CONFIG_VM_EVENT_COUNTERS is not set
 # CONFIG_SLUB_DEBUG is not set
@@ -105,7 +105,6 @@ CONFIG_CRYPTO_AES_ARM64_CE_BLK=y
 CONFIG_JUMP_LABEL=y
 CONFIG_MODULES=y
 CONFIG_MODULE_UNLOAD=y
-CONFIG_MODULE_FORCE_UNLOAD=y
 CONFIG_MODVERSIONS=y
 CONFIG_PARTITION_ADVANCED=y
 CONFIG_CMDLINE_PARTITION=y
@@ -194,6 +193,7 @@ CONFIG_NETFILTER_XT_MATCH_POLICY=y
 CONFIG_NETFILTER_XT_MATCH_PKTTYPE=y
 CONFIG_NETFILTER_XT_MATCH_QUOTA=y
 CONFIG_NETFILTER_XT_MATCH_QUOTA2=y
+CONFIG_NETFILTER_XT_MATCH_QUOTA2_LOG=y
 CONFIG_NETFILTER_XT_MATCH_SOCKET=y
 CONFIG_NETFILTER_XT_MATCH_STATE=y
 CONFIG_NETFILTER_XT_MATCH_STATISTIC=y
@@ -223,8 +223,6 @@ CONFIG_IP6_NF_FILTER=y
 CONFIG_IP6_NF_TARGET_REJECT=y
 CONFIG_IP6_NF_MANGLE=y
 CONFIG_IP6_NF_RAW=y
-CONFIG_IP6_NF_NAT=y
-CONFIG_IP6_NF_TARGET_MASQUERADE=y
 CONFIG_BRIDGE_NF_EBTABLES=y
 CONFIG_BRIDGE_EBT_BROUTE=y
 CONFIG_L2TP=y
@@ -246,6 +244,7 @@ CONFIG_NET_EMATCH_U32=y
 CONFIG_NET_EMATCH_META=y
 CONFIG_NET_EMATCH_TEXT=y
 CONFIG_NET_CLS_ACT=y
+CONFIG_BPF_JIT=y
 CONFIG_BT=y
 CONFIG_BT_RFCOMM=y
 CONFIG_BT_RFCOMM_TTY=y
@@ -258,6 +257,7 @@ CONFIG_BT_HCIUART_H4=y
 CONFIG_RFKILL=y
 CONFIG_DEVTMPFS=y
 CONFIG_FW_LOADER_USER_HELPER=y
+# CONFIG_FW_CACHE is not set
 CONFIG_MALI_MEMORY_GROUP_MANAGER=y
 CONFIG_DMA_CMA=y
 CONFIG_ZRAM=y
@@ -277,6 +277,7 @@ CONFIG_SCSI_SCAN_ASYNC=y
 CONFIG_MD=y
 CONFIG_BLK_DEV_DM=y
 CONFIG_DM_CRYPT=y
+CONFIG_DM_SNAPSHOT=y
 CONFIG_DM_UEVENT=y
 CONFIG_DM_VERITY=y
 CONFIG_DM_VERITY_AVB=y
@@ -284,7 +285,9 @@ CONFIG_DM_VERITY_FEC=y
 CONFIG_DM_BOW=y
 CONFIG_DM_ANDROID_VERITY_AT_MOST_ONCE_DEFAULT_ENABLED=y
 CONFIG_NETDEVICES=y
+CONFIG_DUMMY=y
 CONFIG_TUN=y
+CONFIG_VETH=y
 # CONFIG_NET_VENDOR_3COM is not set
 # CONFIG_NET_VENDOR_ADAPTEC is not set
 # CONFIG_NET_VENDOR_AGERE is not set
@@ -816,12 +819,14 @@ CONFIG_OPTEE=y
 CONFIG_RK_NAND=y
 CONFIG_RK_HEADSET=y
 CONFIG_EXT4_FS=y
+CONFIG_EXT4_FS_POSIX_ACL=y
 CONFIG_EXT4_FS_SECURITY=y
 CONFIG_EXT4_ENCRYPTION=y
 CONFIG_F2FS_FS=y
-# CONFIG_F2FS_FS_POSIX_ACL is not set
 CONFIG_F2FS_FS_SECURITY=y
 CONFIG_F2FS_FS_ENCRYPTION=y
+CONFIG_FS_VERITY=y
+CONFIG_FS_VERITY_BUILTIN_SIGNATURES=y
 CONFIG_QUOTA=y
 CONFIG_QUOTA_NETLINK_INTERFACE=y
 CONFIG_QFMT_V2=y
@@ -853,6 +858,8 @@ CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
 CONFIG_LSM_MMAP_MIN_ADDR=4096
 CONFIG_HARDENED_USERCOPY=y
+CONFIG_STATIC_USERMODEHELPER=y
+CONFIG_STATIC_USERMODEHELPER_PATH=""
 CONFIG_SECURITY_SELINUX=y
 CONFIG_TEE_SUPPORT=y
 CONFIG_CRYPTO_TWOFISH=y
-- 
2.35.3

